common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -138,15 +165,25 @@ protected void serviceInit(Configuration conf) throws Exception {
       registeredExecutorFile =
         new File(getRecoveryPath().toUri().getPath(), RECOVERY_FILE_NAME);
 
+      secretsFile = new File(getRecoveryPath().toUri().getPath(), SECRETS_RECOVERY_FILE_NAME);
```

Comment:
Do you need `secretsFile` declared as a field? It seems to only be used in this method.

Also, it seems all this secret loading code could be done inside an "authEnabled" check, since `reloadCreds` does nothing if auth is disabled anyway.